browser/ui/views/speedreader/speedreader_mode_bubble.cc

```diff
@@ -81,6 +82,7 @@ bool SpeedreaderModeBubble::ShouldShowCloseButton() const {
 
 void SpeedreaderModeBubble::WindowClosing() {
   if (tab_helper_) {
+    tab_helper_->MaybeToggleEnabledForSite(site_toggle_button_->GetIsOn());
```

Comment:
I would consider immediate reload once the toggle is clicked, as we do in shields. For me it wasnot intiutive that I have to close the bubble to reload the page

Reply:
done.